### PR TITLE
[RDY] Return -1 from jobstart when cmd is non executable instead of exception

### PR DESCRIPTION
**Pull request by aarowill**
Addresses the issue here: https://github.com/neovim/neovim/issues/5465 and also specs the proper behavior.

### CODE REVIEW

**Review by jamessan — APPROVED** [review]

File: src/nvim/eval.c [hunk 1]
```diff
@@ -11597,9 +11597,6 @@ static char **tv_to_argv(typval_T *cmd_tv, char **cmd)
   const char_u *exe = get_tv_string_chk(&argl->lv_first->li_tv);
   if (!exe || !os_can_exe(exe, NULL, true)) {
     // String is not executable
```
Comment (justinmk) [inline src/nvim/eval.c]: the comment is not relevant anymore

File: src/nvim/globals.h [hunk 2]
```diff
@@ -1125,7 +1125,6 @@ EXTERN char_u e_invcmd[] INIT(= N_("E476: Invalid command"));
 EXTERN char_u e_isadir2[] INIT(= N_("E17: \"%s\" is a directory"));
 EXTERN char_u e_invjob[] INIT(= N_("E900: Invalid job id"));
 EXTERN char_u e_jobtblfull[] INIT(= N_("E901: Job table is full"));
-EXTERN char_u e_jobexe[] INIT(= N_("E902: \"%s\" is not an executable"));
```
Comment (justinmk) [inline src/nvim/globals.h]: I guess we should remove the E902 cases from the `*.po` files.

File: test/functional/core/job_spec.lua [hunk 3]
```diff
@@ -70,6 +70,11 @@ describe('jobs', function()
     ok(rv ~= nil)
   end)
 
+  it('returns -1 when target is not executable', function()
+    local rv = eval("jobstart(['./test/functional/fixtures/non_executable.txt'])")
+    eq(-1, rv)
```
Comment (justinmk) [inline test/functional/core/job_spec.lua]: I also added a check for `eq("", eval("v:errmsg"))` here.

File: src/nvim/eval.c [hunk 4]
```diff
@@ -11635,6 +11632,7 @@ static void f_jobstart(typval_T *argvars, typval_T *rettv, FunPtr fptr)
 
   char **argv = tv_to_argv(&argvars[0], NULL);
   if (!argv) {
+    rettv->vval.v_number = -1;  // Return -1 on non executable
```
Comment (justinmk) [inline src/nvim/eval.c]: jobstart() doc mentions:

```
Returns:
- 0 on invalid arguments or if the job table is full
- -1 if {cmd}[0] is not executable.
```

`tv_to_argv()` may fail for reasons other than non-executable, such as invalid arguments. So we need more granularity from `tv_to_argv()`.

**Review by justinmk — CHANGES_REQUESTED** [review]

File: src/nvim/eval.c (hunk 4 above)
Comment (justinmk) [inline src/nvim/eval.c]: Fixed in https://github.com/neovim/neovim/pull/5750

**Comment by justinmk** [comment]
Merged in #5750. Thanks @aaron-williamson

**Pull request closed without merging**